### PR TITLE
Update the CLI tutorial with more options

### PR DESCRIPTION
**Pull request by calebrob6**
The CLI is much more useful when it saves the models during training :) -- it also allows you to pick LR scheduler and optimizer without overriding the trainer.

### CODE REVIEW

File: docs/tutorials/cli.ipynb [hunk 1]
```diff
@@ -172,13 +197,99 @@
...
    "metadata": {},
    "source": [
-    "This YAML file has three sections:\n",
+    "This YAML file has five main sections:\n",
+    "\n",
+    "* **trainer**: Arguments to pass to the [Trainer](https://lightning.ai/docs/pytorch/stable/common/trainer.html), including callbacks\n",
+    "* **model**: Arguments to pass to the task\n",
+    "* **data**: Arguments to pass to the data module\n",
+    "* **optimizer**: Custom optimizer configuration\n",
+    "* **lr_scheduler**: Learning rate scheduler configuration\n",
```
Comment (adamjstewart) [inline docs/tutorials/cli.ipynb]: Can we make it clear, either in the YAML file or the comment here, which sections are optional and which are required? I think only model and data are technically required? Don't want to scare away new users for the sake of helping power users.

File: docs/tutorials/cli.ipynb [hunk 2]
```diff
@@ -146,21 +146,46 @@
     "root = os.path.join(tempfile.gettempdir(), 'cli')\n",
     "config = f\"\"\"\n",
     "trainer:\n",
-    "  max_epochs: 1\n",
+    "  max_epochs: 2\n",
     "  default_root_dir: '{root}'\n",
+    "  callbacks:\n",
+    "    - class_path: lightning.pytorch.callbacks.ModelCheckpoint\n",
```
Comment (adamjstewart) [inline docs/tutorials/cli.ipynb:152]: Out of curiosity, does `ModelCheckpoint` without the full path work? What if you add:
```python
import lightning.pytorch.callbacks
```
in `torchgeo/main.py`?

File: docs/tutorials/cli.ipynb [hunk 3]
```diff
@@ -146,21 +146,46 @@
...
     "data:\n",
     "  class_path: EuroSAT100DataModule\n",
     "  init_args:\n",
     "    batch_size: 8\n",
     "  dict_kwargs:\n",
     "    root: '{root}'\n",
     "    download: true\n",
+    "\n",
+    "optimizer:\n",
+    "  class_path: torch.optim.AdamW\n",
```
Comment (adamjstewart) [inline docs/tutorials/cli.ipynb:179]: Same here, we could theoretically add:
```python
import torch.optim
import torch.optim.lr_scheduler
```
to `torchgeo/main.py` if we want things to be more magic.

File: docs/tutorials/cli.ipynb (hunk 2 above)
Comment (calebrob6) [reply docs/tutorials/cli.ipynb:152]: I think ModelCheckpoint without the full path worked, but I want to keep the full path as I don't really like magic

File: docs/tutorials/cli.ipynb (hunk 1 above)
Comment (calebrob6) [reply docs/tutorials/cli.ipynb]: Annotated optimizer and lr_scheduler in this list as optional, keeping the others as is because a user is going to need to change trainer anyway

File: docs/tutorials/cli.ipynb [hunk 4]
```diff
@@ -172,13 +197,99 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "693bb09d",
+   "metadata": {},
+   "source": [
+    "## Optimizer and Learning Rate Scheduler\n",
+    "\n",
+    "By default, TorchGeo trainers use Adam with a ReduceLROnPlateau scheduler. However, you can customize these via the `optimizer` and `lr_scheduler` sections in your config.\n",
```
Comment (adamjstewart) [inline docs/tutorials/cli.ipynb]: We actually default to AdamW, not Adam

File: docs/tutorials/cli.ipynb [hunk 5]
```diff
@@ -172,13 +197,99 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ed5072f5",
+   "metadata": {},
+   "source": [
+    "## Callbacks\n",
+    "\n",
+    "Lightning callbacks allow you to hook into the training loop at various points. In our config, we use two essential callbacks:\n",
```
Comment (adamjstewart) [inline docs/tutorials/cli.ipynb]: I would personally consider EarlyStopping to be essential but LearningRateMonitor to not be essential. In fact, I rarely use any learning rate decay with optimizers like Adam that have dynamic lr built in. Maybe worth swapping these examples?

File: docs/tutorials/cli.ipynb
Comment (adamjstewart) [inline docs/tutorials/cli.ipynb:370]: Do you want to add more links here, for example, to the full list of optimizers/lr schedulers/callbacks?

Comment (calebrob6) [reply docs/tutorials/cli.ipynb:370]: I have that link above "See the [Lightning Callbacks documentation](https://lightning.ai/docs/pytorch/stable/extensions/callbacks.html) for more options."

File: docs/tutorials/cli.ipynb (hunk 5 above)
Comment (calebrob6) [reply docs/tutorials/cli.ipynb]: Changed the "essential" wording as this is just an example

**Review by adamjstewart — APPROVED** [review]
